zebra/debug.c

```diff
@@ -599,21 +594,21 @@ static int config_write_debug(struct vty *vty)
 		}
 	}
 
+	if (IS_ZEBRA_DEBUG_KERNEL_MSGDUMP_SEND
+	    && IS_ZEBRA_DEBUG_KERNEL_MSGDUMP_RECV) {
+		vty_out(vty, "debug zebra kernel msgdump\n");
+		write++;
+	} else if (IS_ZEBRA_DEBUG_KERNEL_MSGDUMP_RECV) {
+		vty_out(vty, "debug zebra kernel msgdump recv\n");
+		write++;
+	} else if (IS_ZEBRA_DEBUG_KERNEL_MSGDUMP_SEND) {
+		vty_out(vty, "debug zebra kernel msgdump send\n");
+		write++;
+	}
+
 	if (IS_ZEBRA_DEBUG_KERNEL) {
-		if (IS_ZEBRA_DEBUG_KERNEL_MSGDUMP_SEND
```

Comment:
related question about this change, since "kernel" looks like the parent of "msgdump"

Reply:
this was never right.  if you set _SEND or _RECV on you never added DEBUG_KERNEL.  You would never see output.